CMakeLists.txt

```diff
@@ -210,7 +210,7 @@ target_link_libraries(dawncpp dawn)
 add_subdirectory(third_party)
 
 add_subdirectory(src/common)
-add_subdirectory(src/backend)
+add_subdirectory(src/libDawn)
```

Comment:
Just so I understand the vision here, what's the plan for the other dirs?

Reply:
It mirrors `libANGLE` in ANGLE. However I'm not strongly attached to this rename myself.

Reply:
Thanks for the bikeshed, maybe we can improve on the ANGLE model. The dependencies of libDawn would be:
 - src/common
 - Some autogenerated files like the C/C++ headers, autogenerated validation etc.
 - SPIR-V Tools stuff like the validator
 - Our SPIR-V cross replacement.

The wire would be in a separate library, maybe libDawnWire because it doesn't need to know about internal details of libDawn. Tests and examples just like against libDawn.

Alternative names could be "implementation", "impl", "dawn", "main". I'd rather not split the frontend and backend in different directories because the nesting of backend namespaces in the frontend's namespace is very useful.

Or maybe something like this:
 - src/
   - common/
   - backend/ maybe with some backend-agnostic utility classes in namespace "dawn_impl::backend"
     - vulkan/ with namespace "dawn_impl::backend::vulkan"
     - other backends
  - frontend/ with namespace "dawn_impl"

@vonture might be able to share what he thinks good have worked better for ANGLE.